### PR TITLE
Login and Registration features completed

### PR DESCRIPTION
**Pull request by jessicatong43**
Created login and registration pages, and both are connected to the back end.

### CODE REVIEW

File: client/src/Components/Register.jsx [hunk 1]
```diff
@@ -1,7 +1,201 @@
...
+
+    // If passwords don't match, user will be notified.
+    if (confPass.length && pass !== confPass) {
+      setPasswordMatch(false);
+    } else if (pass === confPass) {
+      setPasswordMatch(true);
+    }
+
+    // If password is shorter than 6 characters, user will be notified.
+    if (pass.length && pass.length >= 6) {
```
Comment (JMStudiosJoe) [inline client/src/Components/Register.jsx:54]: In the input there is set to 4 & here is set to 6, lets define a const at the top for both to be set by. Is this the only form of validation we have on the password entry? should add either some other checks in regex but that can be another pr.

File: client/src/Components/Register.jsx [hunk 2]
```diff
@@ -1,7 +1,201 @@
...
+        {!passwordMatch
+          && (
+            <div className="matching-password">
+              <small>Passwords do not match</small>
+            </div>
+          )}
+      </form>
+
+      <div className="lower-buttons login-signup-buttons">
+        <button
```
Comment (JMStudiosJoe) [inline client/src/Components/Register.jsx:186]: food for thought on Form tag here, should this button be within the <form tag for sematicness? imo I do like the button instead of <input type submit to "submit" the form details, but im not sure if best to still be inside the form tag or not, can disregard for another pr too.

File: client/src/utils/fetchUserData.js [hunk 3]
```diff
@@ -40,3 +42,20 @@ export async function fetchProfileData(user) {
     console.error(err);
   }
 }
+
+export async function fetchRegister(body) {
```
Comment (JMStudiosJoe) [inline client/src/utils/fetchUserData.js:46]: I am a little confused by the logic of this code block, maybe some comments might help. I think also there might be too much in the one try block when there are multiple things happening to go wrong imo. Would appreciate your thoughts on this.

File: client/src/css/LoginSignUp.css [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+  text-align: center;
+  margin: 0.25rem;
+}
+
+.login-signup-buttons button{
+  width: 50dvw;
+  max-width: 200px;
+}
+
+#login-signup-button {
```
Comment (JMStudiosJoe) [inline client/src/css/LoginSignUp.css:73]: Usually "id" use is for very specifc styling I am wondering why only used here for the signup button when a class would suffice. Likely should be the same styling as most "good" actionable buttons Login, Register, Signup, New Item etc...

File: client/src/Components/Login.jsx [hunk 5]
```diff
@@ -23,42 +31,74 @@ export default function Login({ user, setUser }) {
...
+  const goBack = () => {
+    navigate(-1);
   };
 
   return (
-    <div>
-      <form>
-        <label htmlFor="email">
-          email:
+    <div className="login-signup-container main-container">
```
Comment (JMStudiosJoe) [inline client/src/Components/Login.jsx:39]: This comment applies to both here and Register.jsx to address a few small things. Semantic HTML imo should be more used here tags such as <section> <article> <aside> instead of evertying to be a div. Great use of the form tag and others but the divs, feel a little excessive. <small is well ive never seen it, but its semantic no need to remove haha.

File: client/src/Components/Login.jsx [hunk 6]
```diff
@@ -23,42 +31,74 @@ export default function Login({ user, setUser }) {
...
-    console.log('token: ', token);
-    console.log('userCookie: ', userData);
+  const goBack = () => {
+    navigate(-1);
   };
 
   return (
-    <div>
-      <form>
-        <label htmlFor="email">
```
Comment (JMStudiosJoe) [inline client/src/Components/Login.jsx:37]: why the removal of the labels? labels should be added for accessiblity reasons and clarity. Likely should be a new component of LabelledInput but can be a refactor pr to keep things small imo.

**Review by JMStudiosJoe — CHANGES_REQUESTED** [review]
Fantastic work thank you so much! just a few minor changes needed and some questions to answer but very close to being ready to merge :)

**Review by jessicatong43 — COMMENTED** [review]
Thanks for all your notes so far, Joey! I've answered your questions as best I could, and I also had a few questions of my own.

I have implemented some of your suggestions on my end, and will push my changes up shortly.

File: client/src/Components/Login.jsx (hunk 6 above)
Comment (JMStudiosJoe) [reply client/src/Components/Login.jsx:37]: thats ok i think we will need someone with more insite into accessibility design on this than lil ol me. Thanks for clarifying no changes needed.

File: client/src/utils/fetchUserData.js (hunk 3 above)
Comment (JMStudiosJoe) [reply client/src/utils/fetchUserData.js:52]: yikes ok good to know

**Review by JMStudiosJoe — APPROVED** [review]
